programs/clearing_house/src/controller/position.rs

```diff
@@ -322,7 +375,7 @@ pub fn update_position_with_base_asset_amount(
     mark_price_before: u128,
     now: i64,
     maker_limit_price: Option<u128>,
-) -> ClearingHouseResult<(bool, u128, u128, u128, i128)> {
+) -> ClearingHouseResult<(bool, u128, u128, u128, PositionDelta)> {
```

Comment:
it's weird that you mutate the market here and then have to call the update_position_and_market_with_fee later on. Ideally all the mutation of market state happen more closely together. Can help brainstorm what this architecture might look like. I think it might just mean we need to calculate fees in a different place such that you can update all the market related state at once

Reply:
yea this sequence below is only referred to once in within this function, maybe we make it one larger function?

`calculate_position_delta_with_base_asset_amount`
`calculate_fee_for_order_fulfill_against_amm`
`update_user_and_market_position` (rename of `update_position_and_market_with_fee`)